src/tests/functional/plugin/cpu/single_layer_tests/gru_sequence.cpp

```diff
@@ -42,92 +74,79 @@ class GRUSequenceCPUTest : public testing::WithParamInterface<GRUSequenceCpuSpec
-        std::map<std::string, std::string> additionalConfig;
-
-        size_t seq_lengths;
-        size_t batch;
-        size_t hidden_size;
-        size_t input_size = 10;
+        std::vector<InputShape> inputShapes;
+        ngraph::helpers::SequenceTestsMode seqMode;
         std::vector<std::string> activations;
-        std::vector<float> activations_alpha;
-        std::vector<float> activations_beta;
         float clip;
-        bool linear_before_reset;
-        ngraph::op::RecurrentSequenceDirection direction;
-        InferenceEngine::Precision netPrecision;
+        bool linearBeforeReset;
+        ov::op::RecurrentSequenceDirection direction;
+        ElementType netPrecision;
+        CPUSpecificParams cpuParams;
+        std::map<std::string, std::string> additionalConfig;
 
-        std::tie(basicParamsSet, cpuParams, additionalConfig) = this->GetParam();
+        std::tie(inputShapes, seqMode, activations, clip, linearBeforeReset, direction, netPrecision, cpuParams, additionalConfig) = this->GetParam();
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
-        std::tie(m_mode, seq_lengths, batch, hidden_size, activations, clip, linear_before_reset, direction, netPrecision, targetDevice) = basicParamsSet;
-
-        size_t num_directions = direction == ngraph::op::RecurrentSequenceDirection::BIDIRECTIONAL ? 2 : 1;
-        std::vector<std::vector<size_t>> inputShapes = {
-            {{batch, seq_lengths, input_size},
-             {batch, num_directions, hidden_size},
-             {batch},
-             {num_directions, 3 * hidden_size, input_size},
-             {num_directions, 3 * hidden_size, hidden_size},
-             {num_directions, (linear_before_reset ? 4 : 3) * hidden_size}},
-        };
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+
+        init_input_shapes(inputShapes);
 
         // method MKLDNNMemoryDesc::isSame can't correct compute layout for tensor with strides = 1
         // returned output format always tnc
-        if (inFmts.size() == 2 && ngraph::shape_size(inputShapes[1]) == 1) {
+        if (inFmts.size() == 2 && inputDynamicShapes[1].is_static() && ov::shape_size(inputDynamicShapes[1].to_shape()) == 1) {
             inFmts[1] = tnc;
         }
 
         configuration.insert(additionalConfig.begin(), additionalConfig.end());
 
-        if (additionalConfig[PluginConfigParams::KEY_ENFORCE_BF16] == PluginConfigParams::YES) {
-            inPrc = outPrc = Precision::BF16;
-        } else {
-            inPrc = outPrc = netPrecision;
-        }
+        const size_t hiddenSize = inputDynamicShapes[1][2].get_length();
+        const size_t inputSize = inputDynamicShapes.front()[2].get_length();
+        const size_t numDirections = direction == ov::op::RecurrentSequenceDirection::BIDIRECTIONAL ? 2 : 1;
+
+//        if (additionalConfig[InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16] == InferenceEngine::PluginConfigParams::YES) {
```

Comment:
uncomment please

Reply:
That does not work properly with new tests API.